Tools/encodings revision

graphicore opened the pull request:
This PR builds on top of #642 please review/merge that first.

In this PR I work on the tasks found in #624 #660 and #678

graphicore commented:
@alexeiva I requested a review from you, because I started changing the Namelists and filter lists. see #678  for more details.

alexeiva commented on tools/encodings/GF Glyph Sets/Cyrillic/GF-cyrillic-plus_unique-glyphs.nam:
```diff
@@ -188,4 +191,99 @@
+       descender-cy.case
+       verticalbar-cy
+       verticalbar-cy.case
+       a-cy.sc
```
we shouldn't combine uni- and nice-naming for Cyrillic in one file.
 
ғ uni0493 vs. a-cy.sc

graphicore replied:
Will be reformatted later in the process.

alexeiva commented on tools/encodings/GF Glyph Sets/Cyrillic/GF-cyrillic-pro_unique-glyphs.nam:
```diff
@@ -9,6 +10,8 @@
 0x0495  ҕ uni0495
 0x049E  Ҟ uni049E
 0x049F  ҟ uni049F
+0x04A6  Ҧ CYRILLIC CAPITAL LETTER PE WITH MIDDLE HOOK
```
Glyphs:
```
0x04A6  Ҧ
0x04A7  ҧ
```
have been removed from GF Cyrillic Pro for obsolecy.

graphicore replied:
so they shouldn't be in the filter lists then?

alexeiva replied:
no, not even in filter lists

graphicore replied:
@alexeiva What about these in Cyrillic-Plus:
```
0x0494  Ҕ CYRILLIC CAPITAL LETTER GHE WITH MIDDLE HOOK
0x0495  ҕ CYRILLIC SMALL LETTER GHE WITH MIDDLE HOOK
```

I also added them from the filter-lists.

alexeiva commented on tools/encodings/GF Glyph Sets/Cyrillic/GF-cyrillic-pro_unique-glyphs.nam:
```diff
@@ -1,4 +1,5 @@
 # Created by Alexei Vanyashin 2016-27-06
+#$ include GF-cyrillic-plus_unique-glyphs.nam
 0x048A  Ҋ uni048A
```
@graphicore These lines should be reformatted to `0x | Cha | Unicode Description` format using your latest namelist.py file

graphicore replied:
yeah, I'll do so later. For now it's not so important, because I really only read the unicode codepoint. But I thought about doing so before :-)

alexeiva replied:
Alright. This is good to be merge now then.

alexeiva commented on tools/encodings/GF Glyph Sets/Cyrillic/filter lists/uni names/plus_unique-glyphs.txt:
```diff
@@ -189,4 +191,4 @@ tugrik
 descender-cy
```
Let's move these component-parts from the official list to recommended glyphs. These glyphs are used for component building and are non-exportable. I have seen people forget to set the non-export tag.

```
descender-cy  
descender-cy.case  
verticalbar-cy
verticalbar-cy.case
```

graphicore replied:
Putting these into RECOMMENDED.md with the text:

### Cyrillic component building

These glyphs are used for component building and are non-exportable. Don't forget to set the non-export tag.

```
descender-cy    
descender-cy.case
verticalbar-cy
verticalbar-cy.case

alexeiva commented on tools/encodings/GF Glyph Sets/GF-latin-pro_optional-glyphs.nam:
```diff
@@ -0,0 +1,37 @@
+          guilsinglright.case
+          guillemetleft.case
+          guillemetright.case
+          0x0030 zero.sinf
```
all .sinf glyphs should be removed as per our guide on https://github.com/google/fonts/tree/master/tools/encodings/GF%202016%20Glyph%20Sets#gf-latin-pro-145-for-713-glyphs-total--17-optional-case

graphicore replied:
In https://github.com/google/fonts/issues/675#issuecomment-282672569 we found out that Namelists are parents and filter-lists are derrived from Namelists. In #678 I started writing tests for that. These changes are making the tests pass.

The rule is: if it is in a filter-list it must be also in the according Namelist.

I think we have some room for more advanced rules here. We could demand that it must be either in the `unique` or in the `optional` Namelist. We could also add another optional type of namelist to store components for construction.

alexeiva replied:
_components for construction_ list could be an idea for future. I also have a 'development' filter list of Cyrillic glyphs that require anchors placing prior to generating accented glyphs.

alexeiva commented:
@graphicore thanks for the update. This is good to be merged now. cc @davelab6

graphicore commented:
I didn't update anything yet :-) Thanks for the replies though, I'll see what I can do.

alexeiva commented:
I meant your reply on my review comments. I am fine with this now.

On Mar 13, 2017 18:41, "Lasse Fister" <notifications@github.com> wrote:

> I didn't update anything yet :-) Thanks for the replies though, I'll see
> what I can do.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/google/fonts/pull/685#issuecomment-286146462>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AP3Q6cw8O2jZUTsz-nblcJV_yWdZLEWTks5rlWONgaJpZM4MOzr6>
> .
>

davelab6 commented:
@graphicore please merge when you are ready :)

graphicore commented:
@davelab6  This will include the commits of #642, I guess that's OK.

graphicore commented:
Now, I'd be happy to merge this.

I added a test and the generation of filter lists according to https://github.com/google/fonts/issues/678#issuecomment-284476631

> Also, should we have a test that checks if a Namelist let's say `GF-latin-plus_unique-glyphs.nam` and the filter-list with the exact matching name i.e. `filter lists/plus_unique-glyphs.txt`contain the same set of glyphs? I mean, there are obvious subsets, like `filter lists/plus_unique-glyphs_vietnamese-unique.txt`I wouldn't expect that to be the same set as  `GF-latin-plus_unique-glyphs.nam` but  `filter lists/plus_unique-glyphs.txt` should be an exact match.
>
> And one more thing, we could probably generate some of the filter-lists automatically and so reduce sources of error. I think the mapping via GlyphData.xml that I'm doing is good enough. So once the above issues with the data are solved we could generate nice-names and uni-names  filter-lists for all Namelist files (making the first type of test always pass for these files).

I made all tests pass.

I also added a script to reformat  Namelists to `namelist.py`


@alexeiva do you want to have another look or can I merge?

davelab6 commented:
Thanks!

graphicore commented:
This was a big one :-)

alexeiva commented:
On Mar 13, 2017 21:43, "Lasse Fister" <notifications@github.com> wrote:

*@graphicore* commented on this pull request.
------------------------------

In tools/encodings/GF Glyph Sets/Cyrillic/GF-cyrillic-pro_unique-glyphs.nam
<https://github.com/google/fonts/pull/685#discussion_r105740288>:

> @@ -9,6 +10,8 @@
 0x0495  ҕ uni0495
 0x049E  Ҟ uni049E
 0x049F  ҟ uni049F
+0x04A6  Ҧ CYRILLIC CAPITAL LETTER PE WITH MIDDLE HOOK

@alexeiva <https://github.com/alexeiva> What about these in Cyrillic-Plus:

0x0494  Ҕ CYRILLIC CAPITAL LETTER GHE WITH MIDDLE HOOK
0x0495  ҕ CYRILLIC SMALL LETTER GHE WITH MIDDLE HOOK

I also added them from the filter-lists.


Ghe with middlehook should be kept




You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/google/fonts/pull/685#discussion_r105740288>, or mute
the thread
<https://github.com/notifications/unsubscribe-auth/AP3Q6SCfcA3yEPE36If0kXnLPCFmH05Dks5rlY5ZgaJpZM4MOzr6>
.
